lib/engines/content_block_manager/app/services/content_block_manager/get_preview_content.rb

```diff
@@ -0,0 +1,40 @@
+require "net/http"
+require "json"
+require "uri"
+
+module ContentBlockManager
+  class GetPreviewContent
```

Comment:
I wonder if this class could/should be more of a model-type class, especially as it returns something that looks like a model. I thinking something like:

```ruby
class PreviewContent < Data.define(:title, :html)
  class << self
    def for_content_id(content_id)
        PreviewContent.new(title: content_item["title"], html:)
    end
    
    private

    def content_item
      @content_item ||= begin
        response = Services.publishing_api.get_content(@content_id)
        response.parsed_content
      end
    end

    def frontend_base_path
      Rails.env.development? ? Plek.external_url_for("government-frontend") : Plek.website_root
    end

    def frontend_path
      frontend_base_path + content_item["base_path"]
    end

    def html
      uri = URI(frontend_path)
      @html ||= Nokogiri::HTML.parse(Net::HTTP.get(uri))
    end
  end
end
```

Reply:
good one 👍 liking the use of Data

Reply:
I've done this in a new refactor commit